### ISSUE TITLE
Vehicle does not track weight of passengers correctly

### ISSUE THREAD

**Issue by Tsunder** [CleverRaven/Cataclysm-DDA#21263]
Vehicle does not update its weight:

~~filled a held 200L steel drum while riding a biycyle, (now weighs 200kg more than before)
bicycle is still same mass as it was when the player got on. (170kg)~~

does not update to include the mass of players carried items at all

vehicle should update its mass as its passengers mass change

**Comment by KunstDerFuge** [comment]
I took a look at this one -- the vehicle code was calling a get_weight method on the player, which was only implemented in the Creature base class and would return a generic weight for a creature's body size, so the player's inventory was not accounted for.

I added a new get_weight method in the player class which does account for inventory, so the vehicle code should now automatically use that one instead, if I'm not mistaken. It does build and run, but I'm not sure how to test it. How were you able to get the weight of the vehicle?

It's uploaded on my page under the bugfix-21263-vehicle-weight branch if you'd like to take a look!
